### PR TITLE
efficient-compression-tool: init at 0.9.1

### PR DESCRIPTION
**Pull request by lunik1**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by tpwrules** [comment]
I was able to build and run this on aarch64.

**Comment by lunik1** [comment]
In theory this package should work on darwin too, but I am unable to test it.

File: pkgs/tools/compression/efficient-compression-tool/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+stdenv.mkDerivation rec {
+  pname = "efficient-compression-tool";
+  version = "0.8.3";
+
+  src = fetchFromGitHub {
+    owner = "fhanau";
+    repo = "Efficient-Compression-Tool";
+    rev = "v${version}";
+    sha256 = "sha256-IHqIKVUvis9LtxFAQz7ItBHZ+BxxWD1VKiNmlTV+UyI=";
+    fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/tools/compression/efficient-compression-tool/default.nix:18]:
```suggestion
```
Please use our libpng.

Comment (lunik1) [reply pkgs/tools/compression/efficient-compression-tool/default.nix:18]: I am not sure this is possible: the package uses a customised build of `libpng` (see `pngusr.h`).

Comment (SuperSandro2000) [reply pkgs/tools/compression/efficient-compression-tool/default.nix:18]: It basically disabled support for a lot of things and set those 3 values https://github.com/fhanau/Efficient-Compression-Tool/blob/master/src/pngusr.h#L92-L94

We could just overwrite our libpng. The libpng version used here is 3 years old.

**Comment by lunik1** [comment]
I have updated to the latest version and modified the build system so it uses `libpng` from `nixpkgs`. I don't think this is technically supported, but the resulting binary seems to function.

**Pull request merged**